source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.cc

```diff
@@ -136,6 +143,51 @@ bool LocalRateLimiterImpl::requestAllowed(
   return requestAllowedHelper(tokens_);
 }
 
+uint32_t LocalRateLimiterImpl::maxTokens(
+    absl::Span<const RateLimit::LocalDescriptor> request_descriptors) const {
+  if (!descriptors_.empty() && !request_descriptors.empty()) {
+    for (const auto& request_descriptor : request_descriptors) {
+      auto it = descriptors_.find(request_descriptor);
+      if (it != descriptors_.end()) {
+        return it->token_bucket_.max_tokens_;
+      }
+    }
+  }
+  return token_bucket_.max_tokens_;
+}
+
+uint32_t LocalRateLimiterImpl::remainingTokens(
+    absl::Span<const RateLimit::LocalDescriptor> request_descriptors) const {
+  if (!descriptors_.empty() && !request_descriptors.empty()) {
+    for (const auto& request_descriptor : request_descriptors) {
+      auto it = descriptors_.find(request_descriptor);
+      if (it != descriptors_.end()) {
+        return it->token_state_->tokens_.load(std::memory_order_relaxed);
+      }
+    }
+  }
```

Comment:
This pattern of finding the first matching descriptor seems to be the same for these three functions, so maybe a helper to DRY this up would help readability?

Reply:
Ok, I will add a general method for finding the descriptor.